packages/flutter/lib/src/rendering/sliver_group.dart

```diff
@@ -0,0 +1,123 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+import 'dart:math';
+
+import '../../rendering.dart';
+
+class SliverFlexParentData extends SliverPhysicalParentData with ContainerParentDataMixin<RenderSliver> {
+  int flex = 1;
+  double extent = 0.0;
+}
```

Comment:
Open question here. Should we move `extent` to `SliverPhysicalParentData` so that any widget that takes slivers is allowed to constrain their cross axis (i.e. SliverConstrainedCrossAxis widget is usable anywhere?)

Also currently, I only use the flex value if the provided extent is 0.0, but I think there should be a better way to decide when to use flex vs extent.

Reply:
Flex and extent seem independent from each other, and currently could event conflict with one another, what if both are set?

Reply:
I decided to make them mutually exclusive by adding assert statements in the `performLayout` code.